txbridge/src/main/java/org/jboss/jbossts/txbridge/outbound/BridgeXAResource.java

```diff
@@ -70,7 +71,7 @@
      * @param externalTxId the parent JTA transaction identifier.
      * @param bridgeWrapper the control for the subordinate WS-AT transaction.
      */
-    public BridgeXAResource(Uid externalTxId, BridgeWrapper bridgeWrapper)
+    public BridgeXAResource(Transaction externalTxId, BridgeWrapper bridgeWrapper)
```

Comment:
If you cannot map from Uid to Transaction then you could deprecate the old constructor and throw a java.lang.UnsupportedOperationException with a message that says the method is now deprecated in favour of the other constructor.